3-reveal/migrations/6-MDA/Zambia-2020/deploy/mda_dispense_tasks.psql

```diff
@@ -0,0 +1,221 @@
+-- Deploy reveal_mda_zambia_2020:mda_dispense_tasks to pg
+
+BEGIN;
+
+--SET search_path TO :"schema",public;
+
+-- Compute single-jurisdiction metrics
+CREATE MATERIALIZED VIEW ntd_dispense_task_metrics AS
+WITH metrics AS
+(
+    SELECT
+        tasks.plan_identifier AS plan_id,
+        tasks.group_identifier AS jurisdiction_id,
+        COALESCE(COUNT(latest_events.id) FILTER (WHERE latest_events.form_data->>'nPzqDistribute' IN ('Yes')), 0) AS nPzqDistribute,
+        COALESCE(COUNT(latest_events.id) FILTER (WHERE latest_events.form_data->>'availableForTreatment' IN ('Yes')), 0) AS availableForTreatment,
+        COALESCE(SUM(COALESCE(latest_events.form_data->>'nPzqDistributedQuantity', '0')::DECIMAL), 0) AS nPzqDistributedQuantity
+    FROM tasks
+    LEFT JOIN LATERAL
```

Comment:
is the join order optimized to minimize rows considered? 4th bullet, https://github.com/onaio/reveal-frontend/issues/1289#issuecomment-696320307

Reply:
a little? there's only one join here so I'm not/wasn't sure that the order will matter, but I'm not a pg expert at all.  I thought about pulling the conditions at the bottom into a subquery but figured I could trust postgres to do that optimization.

This was done pretty quickly though, I usually try to avoid lateral joins when I can because it tells PG *how* to do something and gives it less room to optimize.  The other way I considered would be to get the plan/jurisdiction/task/latest_event in a subquery using distinct on an then aggregate on top of that, which would give postgres more options in how it created/filtered the results in that subquery.